lib/lookup-carrier-by-account-lcr.js

```diff
@@ -0,0 +1,42 @@
+const debug = require('debug')('jambonz:db-helpers');
+
+const sqlQueryLcrByAccountSid = `SELECT lcr_sid FROM lcr WHERE account_sid = ? OR
+  service_provider_sid = (SELECT service_provider_sid from accounts where account_sid = ?)`;
+const sqlQueryLcrRouteByLcrSid = 'SELECT * FROM lcr_routes WHERE lcr_sid = ? ORDER BY priority';
+const sqlQueryLcrCarrierSetEntryByLcrRouteSid = `SELECT * FROM lcr_carrier_set_entry 
+  WHERE lcr_route_sid = ? ORDER BY priority`;
+
+/**
+ * Look up voip carrier sid by account sid and number
+ * @param {*} pool
+ * @param {*} logger
+ * @param {*} account_sid
+ * @param {*} number
+ */
+async function lookupCarrierByAccountLcr(pool, logger, account_sid, number) {
+  const pp = pool.promise();
+  try {
+    const [lcrs] = await pp.query(sqlQueryLcrByAccountSid, [account_sid, account_sid]);
+    if (lcrs.length) {
+      const {lcr_sid} = lcrs[0];
+      const [lcr_routes] = await pp.query(sqlQueryLcrRouteByLcrSid, [lcr_sid]);
+      if (lcr_routes.length) {
+        for (const r of lcr_routes) {
+          var matcher = new RegExp(r.regex);
```

Comment:
how does this handle the case where the digit mattern is not a match, but a prefix. For instance, we want the user to be able to enter '44' and it should match UK destinations.